core/src/HIP/Kokkos_HIP_Parallel_Team.hpp

```diff
@@ -425,7 +425,7 @@ __device__ inline void hip_release_scratch_index(int32_t* scratch_locks,
 template <typename FunctorType, typename... Properties>
 class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>, HIP> {
  public:
-  using Policy       = TeamPolicyInternal<HIP, Properties...>;
+  using Policy       = TeamPolicy<Properties...>;
```

Comment:
i guess this is a correction?

Reply:
I don't remember exactly why but I needed this change for the graph to work.

Reply:
just a personal suggestion that maybe this and the other nested typedefs in my other comments above is something that could be done in a separate PR to avoid adding too many drive-by changes to this PR?